tests/Feature/Repositories/RepositoryTestCase.php

```diff
@@ -8,10 +8,22 @@
 
 abstract class RepositoryTestCase extends TestCase
 {
-    protected $repo;
+    private $repo;
 
+    protected function setRepository(BaseRepositoryContract $repo): void
+    {
+        $this->repo = $repo;
+    }
+
+    /**
+     * @return BaseRepositoryContract
+     * @throws \Exception
+     */
     protected function getRepository(): BaseRepositoryContract
     {
+        if ($this->repo === null) {
```

Comment:
Hmm. Is this check actually necessary? That is, wouldn't an exception be thrown, because the return type is mandated to be `BaseRepositoryContract`, if `$this->repo` isn't set?

Reply:
its a more friendly message to get the developer to the right method

Reply:
Okay, that's reasonable.